### PR TITLE
Bulirsch-Stoer Method implementation in ionosphere

### PR DESCRIPTION
**Pull request by kstppd**
I will get back with some plots of the mapping to show comparisons between this and Euler step.

- Also I will do some more cleanup for example in the way we select which method to use, assuming we do keep the Euler step. From what I have seen already this seems to work fine. 

- Also I need to add some guarding for underflow errors and for exceeding stepsizes larger than FSgridSize/2.

- And note to self: Controlling the kOpt parameter -> row where we want to converge in the table used by the Richardson extrapolaton

### CODE REVIEW

File: sysboundary/ionosphere.cpp [hunk 1]
```diff
@@ -768,61 +969,66 @@ namespace SBC {
       };
 
 
-      // Trace node coordinates outwards until a non-sysboundary cell is encountered
-      //#pragma omp parallel for
-      for(uint n=0; n<nodes.size(); n++) {
+      // Pick an initial stepsize
+      Real stepSize = min(100e3, technicalGrid.DX / 2.); 
+      
+      #pragma omp parallel firstprivate(stepSize)
```
Comment (kstppd) [inline sysboundary/ionosphere.cpp:975]: @ursg   Is this thread safe? I am initializing stepSize above as firstprivate and then I am starting a parallel region. In the parallel for below each thread should modify its' own stepSize accordingly.

Comment (ykempf) [reply sysboundary/ionosphere.cpp:975]: Can't you just declare it inside the parallel region? Then it is a variable private to the thread and I don't need to look up `firstprivate`. :)

**Comment by ursg** [comment]
These most recent changes break compilation with the "old" vectorclass version, where the 3d vector is still called Vec3d. In my ionosphere branch, I have fixed this by an appropriate define at the top of ionosphere.cpp: https://github.com/ursg/vlasiator/blob/ionosphere/sysboundary/ionosphere.cpp#L43

Are you ok reverting that change?

**Comment by kstppd** [comment]
Continued  in  #49

**Pull request closed without merging**